common/djangoapps/third_party_auth/provider.py

```diff
@@ -43,9 +43,9 @@ def enabled(cls):
         return sorted(cls._enabled_providers(), key=lambda provider: provider.name)
 
     @classmethod
-    def displayed_for_login(cls):
+    def displayed_for_login(cls, tpa_hint="no_tpa_hint_specified"):
```

Comment:
`provider_id` [is never `None`](https://github.com/edx/edx-platform/blob/99781661a529d293bf7e076be982ee8b8e2213db/common/djangoapps/third_party_auth/models.py#L159-L162) so can we just write this as `def displayed_for_login(cls, tpa_hint=None):` ?

Otherwise this looks great. If you want an official review from me, just add a test then let me know.